src/utilities/literalToValue.js

```diff
@@ -0,0 +1,79 @@
+import { inspect } from '../jsutils/inspect';
+import { invariant } from '../jsutils/invariant';
+import { keyValMap } from '../jsutils/keyValMap';
+
+import { Kind } from '../language/kinds';
+import type { ConstValueNode } from '../language/ast';
+import type { GraphQLInputType } from '../type/definition';
+import {
+  getNamedType,
+  isScalarType,
+  isInputObjectType,
+} from '../type/definition';
+
+/**
+ * Produces a JavaScript value given a GraphQL Value AST.
+ *
+ * A GraphQL type may be provided, which will be used to interpret different
+ * JavaScript values if it defines a `literalToValue` method.
+ *
+ * | GraphQL Value        | JavaScript Value |
+ * | -------------------- | ---------------- |
+ * | Input Object         | Object           |
+ * | List                 | Array            |
+ * | Boolean              | Boolean          |
+ * | String / Enum        | String           |
+ * | Int / Float          | Number           |
+ * | Null                 | null             |
+ *
+ * Note: This function does not perform any type validation or coercion.
+ */
+export function literalToValue(
```

Comment:
similar to other `valueToLiteral` this tries too much: we need a version of this which is definitely type based and does not infer logic from the value: an input object value should never have fields for example which are not defined in the type.